### PR TITLE
Added result set table types

### PR DESCRIPTION
**Pull request by Lisa159**
New prop **showGoToPage** will be added after merging this PR.

### CODE REVIEW

File: lib/src/resultsetTable/types.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  top?: Space;
+  bottom?: Space;
+  left?: Space;
+  right?: Space;
+};
+type Column = {
+  /**
+   * Column display value.
+   */
+  displayValue;
```
Comment (marcialfps) [inline lib/src/resultsetTable/types.ts:12]: React.ReactNode?

File: lib/src/resultsetTable/types.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+};
+type Row = {
+  /**
+   * Value to be displayed in the cell.
+   */
+  displayValue;
```
Comment (marcialfps) [inline lib/src/resultsetTable/types.ts:18]: React.ReactNode?

File: lib/src/resultsetTable/ResultsetTable.tsx [hunk 3]
```diff
@@ -54,15 +54,15 @@ const BothArrows = () => (
   </svg>
 );
 
-const DxcResultsetTable = ({
+const DxcResultsetTable: Overload = ({
```
Comment (marcialfps) [inline lib/src/resultsetTable/ResultsetTable.tsx:57]: Why do u convert JSX.Element to Overload type?

File: lib/src/resultsetTable/types.ts (hunk 1 above)
Comment (Lisa159) [reply lib/src/resultsetTable/types.ts]: Done

File: lib/src/resultsetTable/types.ts (hunk 2 above)
Comment (Lisa159) [reply lib/src/resultsetTable/types.ts]: Done

File: lib/src/resultsetTable/ResultsetTable.tsx (hunk 3 above)
Comment (Lisa159) [reply lib/src/resultsetTable/ResultsetTable.tsx]: Removed pattern.

File: lib/src/resultsetTable/types.ts [hunk 4]
```diff
@@ -9,28 +9,34 @@ type Column = {
...
-  isSortable: boolean;
-};
-type RowSortable = Row & {
+  displayValue: React.ReactNode;
   /**
    * Value to be used when sorting the table by that
    * column. If not indicated displayValue will be used for sorting.
    */
-  sortValue: boolean;
+  sortValue?: boolean;
```
Comment (marcialfps) [inline lib/src/resultsetTable/types.ts:27]: This prop is not boolean, I think it must be `string` type.

**Review by marcialfps — APPROVED** [review]

**Pull request merged**